tools/test_check_spec_duplication.py

```
@@ -0,0 +1,122 @@
+        path = root / rel
+        text = path.read_text(encoding="utf-8")
+        if text.count(old) < 1:
+            raise SystemExit(f"test setup broken: {old!r} not in {rel}")
+        path.write_text(text.replace(old, new, 1 if once else -1), encoding="utf-8")
+
+    return mutate
+
+
+# (name, mutate(root) -> None, expected_exit)
+CASES: list[tuple[str, Callable[[Path], None], int]] = [
+    ("unmodified tree", lambda root: None, 0),
```

Ruff (0.16.3)

[warning] Unused lambda argument: `root`

(ARG005)